src/Installer/Microsoft.Dotnet.Installation/Internal/DotnetArchiveExtractor.cs

```diff
@@ -9,80 +9,49 @@
 using System.Linq;
 using System.Runtime.InteropServices;
 using Microsoft.Deployment.DotNet.Releases;
+using Microsoft.DotNet.NativeWrapper;
 
 namespace Microsoft.Dotnet.Installation.Internal;
 
-internal class ArchiveDotnetExtractor : IDisposable
+internal class DotnetArchiveExtractor : IDisposable
 {
     private readonly DotnetInstallRequest _request;
     private readonly ReleaseVersion _resolvedVersion;
+    private readonly ReleaseManifest _releaseManifest;
     private readonly IProgressTarget _progressTarget;
     private string scratchDownloadDirectory;
     private string? _archivePath;
 
-    public ArchiveDotnetExtractor(DotnetInstallRequest request, ReleaseVersion resolvedVersion, IProgressTarget progressTarget)
+    public DotnetArchiveExtractor(DotnetInstallRequest request, ReleaseVersion resolvedVersion, ReleaseManifest releaseManifest, IProgressTarget progressTarget)
     {
         _request = request;
         _resolvedVersion = resolvedVersion;
+        _releaseManifest = releaseManifest ?? new();
         _progressTarget = progressTarget;
         scratchDownloadDirectory = Directory.CreateTempSubdirectory().FullName;
     }
 
     public void Prepare()
     {
-        using var releaseManifest = new ReleaseManifest();
+        using var archiveDownloader = new DotnetArchiveDownloader(_releaseManifest);
         var archiveName = $"dotnet-{Guid.NewGuid()}";
         _archivePath = Path.Combine(scratchDownloadDirectory, archiveName + DnupUtilities.GetArchiveFileExtensionForPlatform());
 
         using (var progressReporter = _progressTarget.CreateProgressReporter())
         {
             var downloadTask = progressReporter.AddTask($"Downloading .NET SDK {_resolvedVersion}", 100);
             var reporter = new DownloadProgressReporter(downloadTask, $"Downloading .NET SDK {_resolvedVersion}");
-            var downloadSuccess = releaseManifest.DownloadArchiveWithVerification(_request, _resolvedVersion, _archivePath, reporter);
+            var downloadSuccess = archiveDownloader.DownloadArchiveWithVerification(_request, _resolvedVersion, _archivePath, reporter);
             if (!downloadSuccess)
             {
                 throw new InvalidOperationException($"Failed to download .NET archive for version {_resolvedVersion}");
             }
 
+            Console.WriteLine($"Download of .NET SDK {_resolvedVersion} complete.");
```

Comment:
In the library, I don't think we should have any direct console output.  If we do want to log messages it should be via an abstraction.